agent/handlers/taskmetadata/helpers.go

```diff
@@ -29,7 +29,7 @@ const (
 type errorMessage struct {
 	Code          string `json:"code"`
 	Message       string `json:"message"`
-	httpErrorCode int    `json:"-"`
+	HttpErrorCode int    `json:"-"`
```

Comment:
I don't think we need to export it.  `json:"-"` should ignore the field when marshaling.  If you leave it unexported you can remove the `json` tag.